### PR TITLE
feat(cohorts): cohorts from HogQLQuery

### PR DESCRIPTION
**Pull request by mariusandra**
## Problem

Followup to https://github.com/PostHog/posthog/pull/25310

## Changes

When you have a HogQL query selecting either `persons_id`, `actor_id` or `id`, or a query selecting anything from `events` or `persons`, you can now turn that into a static cohort:

This also includes the SQL behind other queries like the "activity" page:

The caveat is that it has to be a saved insight, though we can definitely add this to a few other "export" buttons.

## How did you test this code?

Added tests

### CODE REVIEW

**Review by aspicer — APPROVED** [review]
This is awesome.

I will likely look at expanding this functionality to create dynamic cohorts from saved hogql queries in the near future.

**Comment by mariusandra** [comment]
@aspicer ready for a re-review, otherwise I'll consider it ticked and merge tomorrow

File: posthog/models/cohort/util.py [hunk 1]
```diff
@@ -74,12 +75,36 @@ def format_person_query(cohort: Cohort, index: int, hogql_context: HogQLContext)
...
+                columns[expr.alias] = expr.expr
+            elif isinstance(expr, ast.Field):
+                columns[str(expr.chain[-1])] = expr
+        column: ast.Expr | None = columns.get("person_id") or columns.get("actor_id") or columns.get("id")
+        if isinstance(column, ast.Alias):
+            select_query.select = [ast.Alias(expr=column.expr, alias="actor_id")]
+        elif isinstance(column, ast.Field):
+            select_query.select = [ast.Alias(expr=column, alias="actor_id")]
+        else:
+            # Support the most common use cases
```
Comment (aspicer) [inline posthog/models/cohort/util.py:99]: How nice of you

Comment (mariusandra) [reply posthog/models/cohort/util.py:99]: That's the lazy option. The more complete alternative is to actually go into the fields of all the joined tables, and to see if any of them link to persons. That might be over thinking it though 😄

**Pull request merged**